index.js

```diff
@@ -99,7 +104,51 @@ async function pay (plugin, {
   }
 }
 
+async function pull (plugin, {
+  pointer,
+  amount,
+  streamOpts = {}
+}) {
+  await plugin.connect()
+  const receiveMax = amount || Infinity
+
+  const response = await query(pointer)
+
+  if (response.contentType.indexOf('application/spsp4+json') !== -1) {
+    const ilpConn = await createConnection({
+      plugin,
+      destinationAccount: response.destinationAccount,
+      sharedSecret: response.sharedSecret,
+      ...streamOpts
+    })
+
+    const stream = await ilpConn.createStream()
+
+    try {
+      await stream.receiveTotal(receiveMax, { timeout: streamOpts.timeout })
+    } catch (err) {
+      const totalReceived = stream.totalReceived
+      try{
+        await ilpConn.end()
+      } catch (err) {
+        logger.debug('Idle timer expired before response packet was received when trying to end connection.')
```

Comment:
It might be possible for some other kind of error to happen (not related to the idle timer), so maybe make this message more generic like:
```suggestion
        logger.debug('Error while ending connection:', err)
```